File: app/src/main/java/com/example/secondhandmarket/CreateActivity.java
package com.example.secondhandmarket;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.example.secondhandmarket.databinding.ActivityLoginBinding;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class CreateActivity extends AppCompatActivity {
    private FirebaseAuth mFirebaseAuth;
    private DatabaseReference mDatabaseRef;
    private EditText mEtEmail, mEtPwd, mEtName, mEtBirth;
    private Button mBtnRegister;

    @SuppressLint("MissingInflatedId")
    protected void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_create);

        mFirebaseAuth = FirebaseAuth.getInstance();
        mDatabaseRef = FirebaseDatabase.getInstance().getReference();

        mEtEmail = findViewById(R.id.createemail);
        mEtPwd = findViewById(R.id.createpwd);
        mEtName = findViewById(R.id.cname);
        mEtBirth = findViewById(R.id.cbirth);
        mBtnRegister = findViewById(R.id.realCreateBtn);

        mBtnRegister.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v){
                String strEmail = mEtEmail.getText().toString();
                String strPwd = mEtPwd.getText().toString();
                String strName = mEtName.getText().toString();
                String strBirth = mEtBirth.getText().toString();

                mFirebaseAuth.createUserWithEmailAndPassword(strEmail, strPwd).addOnCompleteListener(CreateActivity.this, new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if (task.isSuccessful()){
                            FirebaseUser firebaseUser = mFirebaseAuth.getCurrentUser();
                            UserAccount account = new UserAccount();
                            account.setIdToken(firebaseUser.getUid());
                            account.setEmailId(firebaseUser.getEmail());
                            account.setPassword(strPwd);
                            account.setName(strName);
                            account.setBirth(strBirth);

                            mDatabaseRef.child("UserAccount").child(firebaseUser.getUid()).setValue(account);

                            Toast.makeText(CreateActivity.this, "회원가입 성공!", Toast.LENGTH_SHORT).show();
                            Intent intent = new Intent (CreateActivity.this, LoginActivity.class);
                            startActivity(intent);
                        }else {
                            Toast.makeText(CreateActivity.this, "회원가입 실패!", Toast.LENGTH_SHORT).show();
                        }
                    }
                });

            }
        });
    }

}

File: app/src/main/java/com/example/secondhandmarket/CreateActivity.kt
@file:Suppress("DEPRECATION")

package com.example.secondhandmarket

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.secondhandmarket.databinding.ActivityLoginBinding
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase

class CreateActivity : AppCompatActivity() {
    private lateinit var mFirebaseAuth: FirebaseAuth
    private lateinit var mDatabaseRef: DatabaseReference
    private lateinit var mEtEmail: EditText
    private lateinit var mEtPwd: EditText
    private lateinit var mEtName: EditText
    private lateinit var mEtBirth: EditText
    private lateinit var mBtnRegister: Button
    private lateinit var backButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_create)

        mFirebaseAuth = FirebaseAuth.getInstance()
        mDatabaseRef = FirebaseDatabase.getInstance().reference

        mEtEmail = findViewById(R.id.createemail)
        mEtPwd = findViewById(R.id.createpwd)
        mEtName = findViewById(R.id.cname)
        mEtBirth = findViewById(R.id.cbirth)

        backButton = findViewById(R.id.backbutton)
        backButton.setOnClickListener{
            val intent = Intent(this, LoginActivity::class.java)
            startActivity(intent)
        }
        mBtnRegister = findViewById(R.id.realCreateBtn)
        mBtnRegister.setOnClickListener {
            val strEmail = mEtEmail.text.toString()
            val strPwd = mEtPwd.text.toString()
            val strName = mEtName.text.toString()
            val strBirth = mEtBirth.text.toString()

            mFirebaseAuth.createUserWithEmailAndPassword(strEmail, strPwd)
                .addOnCompleteListener(this) { task ->
                    if (task.isSuccessful) {
                        val firebaseUser = mFirebaseAuth.currentUser
                        val account = UserAccount().apply {
                            idToken = firebaseUser?.uid
                            emailId = firebaseUser?.email
                            password = strPwd
                            name = strName
                            birth = strBirth
                        }

                        mDatabaseRef.child("UserAccount").child(firebaseUser?.uid ?: "").setValue(account)

                        Toast.makeText(this, "회원가입 성공!", Toast.LENGTH_SHORT).show()
                        val intent = Intent(this, LoginActivity::class.java)
                        startActivity(intent)
                    } else {
                        Toast.makeText(this, "회원가입 실패!", Toast.LENGTH_SHORT).show()
                    }
                }
        }
    }

}